internal/rename/connector/executor.go

```diff
@@ -0,0 +1,54 @@
+/*
+ *  *******************************************************************************
+ *  * Copyright (c) 2019 Edgeworx, Inc.
+ *  *
+ *  * This program and the accompanying materials are made available under the
+ *  * terms of the Eclipse Public License v. 2.0 which is available at
+ *  * http://www.eclipse.org/legal/epl-2.0
+ *  *
+ *  * SPDX-License-Identifier: EPL-2.0
+ *  *******************************************************************************
+ *
+ */
+
+package connector
+
+import (
+	"fmt"
+
+	"github.com/eclipse-iofog/iofog-go-sdk/pkg/client"
+	"github.com/eclipse-iofog/iofogctl/internal/config"
+	"github.com/eclipse-iofog/iofogctl/pkg/util"
+)
+
+func Execute(namespace, name, newname string) error {
+	// Check that Connector exists in current namespace
+	_, err := config.GetConnector(namespace, name)
+	if err != nil {
+		return err
+	}
+
+	util.SpinStart(fmt.Sprintf("Renaming Connector %s", name))
+
+	// Do a shallow rename of controller
+	ns, err := config.GetNamespace(namespace)
+	if err != nil {
+		return err
+	}
+	for i, cnct := range ns.Connectors {
```

Comment:
Same comment as Agent executor. You should not be looping here to perform an edit into config types. There is an API for this e.g. config.UpdateConnector(). You already have the connector struct to update above